components/tidb_query_expr/src/lib.rs

```diff
@@ -91,10 +92,51 @@ fn map_compare_in_string_sig(ret_field_type: &FieldType) -> Result<RpnFnMeta> {
     })
 }
 
-fn map_like_sig(ret_field_type: &FieldType) -> Result<RpnFnMeta> {
+fn map_like_sig(ret_field_type: &FieldType, children: &[Expr]) -> Result<RpnFnMeta> {
+    let target_collation = children[0]
+        .get_field_type()
+        .as_accessor()
+        .collation()
+        .map_err(tidb_query_datatype::codec::Error::from)?;
+    let pattern_collation = children[1]
+        .get_field_type()
+        .as_accessor()
+        .collation()
+        .map_err(tidb_query_datatype::codec::Error::from)?;
+
     Ok(match_template_collator! {
         TT, match ret_field_type.as_accessor().collation().map_err(tidb_query_datatype::codec::Error::from)? {
-            Collation::TT => like_fn_meta::<TT>()
+            Collation::TT =>  match_template_collator!(
```

Comment:
Can we make it more readable?

Reply:
I add a new macro to make it into only one layer :thinking: 

(Though I don't know whether it's a good habit to hide the complexity into macro...)

Reply:
See https://github.com/tikv/tikv/pull/13785/files#diff-2e5b75759b522e863db26feb08e72847c515f0e04e69d5105294931a1d25751aR45